solution/index.js

```diff
@@ -0,0 +1,758 @@
+    if(liTask.parentElement.id === "todo-tasks"){
+        //delete todo task from data object
+        data["todo"].forEach( (taskObj, i) => taskObj.task === liTask.firstChild.textContent
+                                                                    && taskObj.date === liTask.childNodes[1].childNodes[1].textContent 
+                                                                    && taskObj.time === liTask.childNodes[1].childNodes[2].textContent
+                                                                    ?  data["todo"].splice(i,1)
+                                                                    :  console.log(taskObj) ); 
+    }
+    
+
+    //remove in-progress tasks from data collectors
+    if(liTask.parentElement.id === "in-progress-tasks") {
+        //delete in-progress task from data object
+        data["inProgress"].forEach( (taskObj, i) => taskObj.task === liTask.firstChild.textContent 
+                                                                    && taskObj.date === liTask.childNodes[1].childNodes[1].textContent
+                                                                    && taskObj.time === liTask.childNodes[1].childNodes[2].textContent
+                                                                    ?  data["inProgress"].splice(i,1) 
+                                                                    :  console.log(taskObj));
+    }
+    
+
+    //remove done tasks from data collectors
+    if(liTask.parentElement.id === "done-tasks"){
+        //delete done task from data object
+        data["done"].forEach( (taskObj, i) => taskObj.task === liTask.firstChild.textContent
+                                                                    && taskObj.date === liTask.childNodes[1].childNodes[1].textContent 
+                                                                    && taskObj.time === liTask.childNodes[1].childNodes[2].textContent 
+                                                                    ?  data["done"].splice(i,1)
+                                                                    :  console.log(taskObj) );
+    }
+
+     //remove from localstorage todo === add updated data boject
+     setDataToLocalStorage();
+    
+
+ }
+    
+
+    
+
+    
+
+
+
+    // listItems.forEach((list) => { //observing the changes for every list
+    // observer.observe( list, {childList: true }); // observing for list items which being added
+    // });
+/** ~~ !!! STOP THE MUTANT WHEN USER NOT MAKING AN EVENT RELATED TO IT!!! ~~ */
+
+document.addEventListener( 'keydown', (e) => { // detectin keyboard events are only on input type elements so i detect it through window object
+    
+    if( e.altKey || e.altGraphKey )
+    {   
+        e.preventDefault(); //preventing default so it wont switch to other objects and wont capture the windows events!
+
+        if(isHover && e.key === '1' && currentHoveredEl.parentElement.parentElement.parentElement.id !== "todo-tasks-section")    
```

Comment:
If you had put the event handler on the list items you wouldn't have had a need for this `parentElement` celebration

Reply:
Don't use global variables. It always creates problems. You could indicate a list item is hovered by adding a class to the DOM element, for example

Reply:
Why do you care about if the task already resides in the list it is moved to?

Reply:
Becuz it's dumb to move an item which already in the list to the current list.. so I just prevent it from happening... so it will work only if user try to move it to another list... which is more sense making

Reply:
Yeah I understood your original purpose, but it's good to avoid making optimizations before they are actually needed, especially when that makes the code less generic and readable.
Besides, according to the project requirements, moving a task between lists should set it at the top of the target list. So in fact when you move a task to the same list it does make a difference, because it should move to the top.

Reply:
Ye I know that, but still why move it to the top... instead I thought I would make the drag and drop thing which he can move to whenever he wants to... but didn't make it

Reply:
Because that's the requirement :)

Reply:
Ye it's moving everything to the top...